alerting.md

```diff
@@ -0,0 +1,109 @@
+
+## How to setup an alert?
+
+### Check your alert is valid
+
+Why do we need the alert? A good answer would be: "Because if we don't alert on `<insert your reason here>` users won’t be able to book and we are going to lose money!"
+
+Another scenario worth considering: "Imagine it’s Saturday night and your feature is breaking. Do you need to wake up the Duty engineer or can we live with it being broken until Monday morning?"
+
+Last but not least: "Is the step you want to monitor on the happy path? Should it be?"
+
+### Alert on symptoms, not causes
+
+Symptom based alerting is focused around creating rules and conditions that mimic those of an end user. Setting the alert on the symptom allows you to catch multiple causes with a single alert rule.
+
+Users care about a feature not working not why it’s not working – users care about a symptom not cause. For example an user might be upset when:
+* Page loads are slow
+* There are errors on page or page does not load completely
+* Invalid data is displayed
+* “A button is not working”
+
+Example:
+The end user does not care about MySQL server being unreachable. She cares about not being able to view a property or sign in.
+
+Example 2: Less extreme example would be users not receiving transactional emails. There might be multiple causes why (code error, delayed jobs not processing, Mandrill not accessible, ...). Alerting on cause would mean setting up alerts for each cause.
+
+### Peer review for all alerts
+
+Every added alert should be reviewed by another engineer. Think of it as a PR for the new alert. This ensures knowledge is shared and an extra pair of eyes may spot some obvious errors.
+
+## Tools and services
+
+Quick overview of the tools we use to detect abnormal behaviour and notify on alerts.
+
+### Monitoring the alert
+
+After the alert has been put to production you have ownership of it. This means you have to track its accountability. When it’s raised you should answer these questions:
+Is the alert catching the symptom?
+Was the alert raised in timely fashion?
+Was it a genuine alert (signal) and not a fake one (noise)?
+Did you take any actions in order to fix it?
+
+	If all above questions are answered affirmatively, congratulations! If not, iterate!
+
+### New Relic (NR)
+
+Pros:
+* Setting up response times and errors per user action (i.e. high error rate on user registration)
+* Pinging an application (pingdom-like health check, checks if the application is online and responds with success)
+* Tracking response times on the web server
+
+Cons:
+* Very sensitive when monitoring end user response times (slow clients)
+* Possible false alerts when low traffic (i.e. 3 requests in last 5 minutes, ⅔ were errors => 66% error rate)
+
+### DataDog (DD)
+
+Pros:
+* Background process healthcheck (i.e. scheduler is working)
+* Ensuring background jobs are being processed (i.e. properties being imported, rates synced)
+* Define custom alert required for the business (i.e.
```

Comment:
Good catch. I guess that's what happens when you stare at the same screen for too long :smile: